### DISCUSSION TITLE
Is it able for DPgen to Connect to Internal Servers via Double SSH Jump?

### DISCUSSION THREAD

**Discussion by KaiservonAfrika** [deepmodeling/dpgen#1696] [Q&A]
**Summary**
I've encountered a situation where I need to connect to an internal server that can only be accessible via a double SSH jump (i.e., I first SSH into an intermediate server, and then SSH into the target internal server from there).

The intermediate server is the login server in our university, thus running DPgen directly on it is forbidden. The target server is not in any queue system, I have to use local batch on it.

After reviewing the documentation and experimenting with the configuration options, I couldn't find a way to set up the software to handle this type of connection. I tried to add` "execute_command":"ssh -i id_rsa user@ip"` which works on the intermediate server, to `remote_profile` section in `machine.json`, but this does not work. I was wondering if this functionality is supported in the current version of the software, or if it's planned for a future release.

If this functionality is indeed supported, could you please provide some guidance or examples on how to configure the software to connect to an internal server via a double SSH jump? Thank you very much.

**DP-GEN Version**
dpgen ==0.13.0

**Comment by njzjz** [comment]
https://github.com/deepmodeling/dpdispatcher/pull/493 should be on the right direction, although that PR is stale.
